.travis.yml

```diff
@@ -62,89 +86,128 @@ matrix:
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="1/4"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="2/4"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="3/4"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="4/4"
+        - FASTCACHE="false"
+
   allow_failures:
-    - python: 3.3 # Dummy value
+    - python: "pypy"
       env:
-        - TEST_SAGE="true"
+        - TEST_DOCTESTS="true"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="1/4"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="2/4"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="3/4"
         - FASTCACHE="false"
-      sudo: true
     - python: "pypy"
       env:
+        - TEST_SYMPY="true"
         - SPLIT="4/4"
+        - FASTCACHE="false"
+
 before_install:
   - if [[ "${FASTCACHE}" != "false" ]]; then
       pip install fastcache;
     fi
   - if [[ "${TEST_AUTOWRAP}" == "true" ]]; then
       pip uninstall -y numpy;
-      sudo apt-get update;
-      sudo apt-get install gfortran gcc python-numpy cython;
     fi
   - if [[ "${TEST_GMPY}" == "true" ]]; then
       pip install "gmpy==1.16";
     fi
   - if [[ "${TEST_SPHINX}" == "true" ]]; then
-      sudo apt-get update;
-      sudo apt-get install --no-install-recommends graphviz inkscape texlive texlive-xetex texlive-fonts-recommended texlive-latex-extra lmodern librsvg2-bin imagemagick docbook2x;
       pip install "sphinx==1.3.1";
     fi
-  - if [[ "${TEST_MATPLOTLIB}" == "true" ]]; then
-      pip install "numpy==1.7.1";
-      pip install --allow-external "matplotlib==1.2.1";
+  - |
+    if [[ "${TEST_MATPLOTLIB}" == "true" ]]; then
+    # We do this conditionally because it saves us some downloading if the
+    # version is the same.
+        if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
+          wget https://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O miniconda.sh;
+        else
+          wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+        fi
+        bash miniconda.sh -b -p $HOME/miniconda;
+        export PATH="$HOME/miniconda/bin:$PATH";
+        hash -r;
+        conda config --set always_yes yes --set changeps1 no;
+        conda update -q conda;
+        conda info -a;
+        conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION matplotlib numpy scipy pip;
+        source activate test-environment;
     fi
   - if [[ "${TEST_THEANO}" == "true" ]]; then
-      pip install "numpy==1.7.1";
-      pip install "scipy==0.12.0";
-      pip install --no-deps https://github.com/Theano/Theano/archive/rel-0.6.zip;
+      pip install "theano";
```

Comment:
It doesn't really matter (it only takes 4 seconds to pip install). I failed to notice that it has a conda build.